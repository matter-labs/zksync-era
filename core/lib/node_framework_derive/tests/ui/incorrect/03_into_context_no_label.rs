#![allow(dead_code)]

use zksync_node_framework_derive::IntoContext;

mod service {
    pub struct ServiceContext<'a> {
        _marker: std::marker::PhantomData<&'a ()>,
    }

    impl<'a> ServiceContext<'a> {
        pub fn add_task<T>(&mut self, _task: T) -> &mut Self {
            unimplemented!()
        }

        pub fn insert_resource<T>(&mut self, _resource: T) -> Result<(), super::WiringError> {
            unimplemented!()
        }
    }
}

pub enum WiringError {
    ResourceLacking { _name: String },
}

pub trait IntoContext {
    fn into_context(self, context: &mut service::ServiceContext<'_>) -> Result<(), WiringError>;
}

#[derive(IntoContext)]
#[ctx(local)]
struct SimpleStruct {
    #[resource]
    _field: u8,
    _field_2: u16,
}

fn main() {}
