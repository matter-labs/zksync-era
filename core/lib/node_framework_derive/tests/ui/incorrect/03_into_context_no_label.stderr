error: All fields must have either `task` or `resource` attribute
  --> tests/ui/incorrect/03_into_context_no_label.rs:34:5
   |
34 |     _field_2: u16,
   |     ^^^^^^^^
